servant/src/Servant/Links.hs

```diff
@@ -661,19 +661,19 @@ simpleToLink _ toA _ = toLink toA (Proxy :: Proxy sub)
 
 -- Erroring instance for 'HasLink' when a combinator is not fully applied
 instance TypeError (PartialApplication
-
+#if __GLASGOW_HASKELL__ >= 904
                     @(Type -> Constraint)
-
+#endif
                     HasLink arr) => HasLink ((arr :: a -> b) :> sub)
   where
     type MkLink (arr :> sub) _ = TypeError (PartialApplication (HasLink :: Type -> Constraint) arr)
     toLink = error "unreachable"
 
 -- Erroring instances for 'HasLink' for unknown API combinators
 instance {-# OVERLAPPABLE #-} TypeError (NoInstanceForSub
-
+#if __GLASGOW_HASKELL__ >= 904
                                          @(Type -> Constraint)
-
+#endif
```

Comment:
why 904?

Reply:
You're watching the second commit, which reinstates these lines after they were mistakenly removed in the first commit.